File: py/movieGenre.py
class Movie:
    def __init__(self):
        self.uuid = ""
        self.title = ""
        self.year = ""
        self.genres = None
        self.related = []

    def addRelatedMovie(self, m):
        if isinstance(m, Movie):
            related.append(m)
            return true
        return false

    def setGenre(self, g):
        if isinstance(g, Genre):
            genres = g
            g.addMovie(self)
            return true
        return false

class Genre:
    def __init__(self):
        self.name = ""
        self.movies = []

    def addMovie(self, m):
        if isinstance(m, Movie):
            self.movies.append(m)
            return true
        return false

File: js/movieGenre.js
class Movie {
    constructor(){
        this.uuid = "";
        this.title = "";
        this.year = "";
        this.genres = null;
        this.related = [];   
    }
   addRelatedMovie(m){
        if(m instanceof Movie){
            this.related.push(m);
            return true;
        }
        return false;
    }
    setGenre(g){
        if(g instanceof Genre){
           this.genres = g;
           g.addMovie(this);
           return true;
        }
        return false;
    }

}

class Genre{
    constructor(){
        this.name = "";
        this.movies = [];
    }
    addMovie(m){
        if(m instanceof Movie){
            this.movies.push(m);
            return true;
        }
        return false;
    }
}


